static/js/click_handlers.js

```diff
@@ -447,11 +447,22 @@ export function initialize() {
 
     $("body").on("click", "#recent_topics_table .on_hover_topic_read", (e) => {
         e.stopPropagation();
-        const topic_row_index = $(e.target).closest("tr").index();
+        const $elt = $(e.currentTarget);
+        const topic_row_index = $elt.closest("tr").index();
         recent_topics_ui.focus_clicked_element(topic_row_index, recent_topics_ui.COLUMNS.read);
-        const stream_id = Number.parseInt($(e.currentTarget).attr("data-stream-id"), 10);
-        const topic = $(e.currentTarget).attr("data-topic-name");
-        unread_ops.mark_topic_as_read(stream_id, topic);
+        if (e.currentTarget.classList.contains("unread_count_pm")) {
+            // PM row
+            const user_ids_string = $elt.attr("data-user-ids-string");
+            unread_ops.mark_pm_as_read(user_ids_string);
+        } else {
+            // Stream row
+            const stream_id = Number.parseInt($elt.attr("data-stream-id"), 10);
+            const topic = $elt.attr("data-topic-name");
+            unread_ops.mark_topic_as_read(stream_id, topic);
+        }
+        // Ideally, we want to send user to the row below but that seems
+        // dangerous since a continuous `enter` press could mark all rows as read.
+        recent_topics_ui.change_focused_element($elt, "right_arrow");
```

Comment:
I think holding down `Enter` isn't a super likely situation; so I think it would actually be better to have it leave you at the next topic; that would make it more ergonomic to mark several things as read after sorting via this screen.

Reply:
(But we if the next row has no "Unread" column, we should go to the topic column, not the mute one, since accidental muting would be annoying)

Reply:
That seems reasonable to me... Probably more useful for the Inbox view than here.